odo/convert.py

```diff
@@ -195,7 +196,20 @@ def list_to_numpy(seq, dshape=None, **kwargs):
     if (seq and isinstance(seq[0], Iterable) and not ishashable(seq[0]) and
             not isscalar(dshape)):
         seq = list(map(tuple, seq))
-    return np.array(seq, dtype=dshape_to_numpy(dshape))
+    try:
+        return np.array(seq, dtype=dshape_to_numpy(dshape))
+    except UnicodeEncodeError:
+        ds = dshape_to_numpy(dshape)
+        for i, rec in enumerate(seq):
+            if sys.version_info.major == 3:
```

Comment:
it looks like the only difference in these branches is the `str` vs `unicode` call. We could just assign a `string_type` function in this branch and then share the expression in both cases.